project/frontend/src/components/ProfileForm.js

```diff
@@ -1,45 +1,107 @@
-import React from 'react';
 import './ProfileForm.css';
 
-import PreferenceForm from './PreferenceForm.js';
+import React, { useContext, useState } from 'react';
+
+import { AuthContext } from '../components/Authentication.js';
 import ButtonGroup from './ButtonGroup';
+import FirestoreContext from '../contexts/FirestoreContext.js';
+import PreferenceForm from './PreferenceForm.js';
 
 /**
  * A form used to gather information for a user's profile page.
  */
 function ProfileForm() {
+  const { firestore } = useContext(FirestoreContext);
+  const authContext = useContext(AuthContext);
+  let user = authContext.currentUser.get;
+
   const itemLabels = {
     cuisine: 'Preferred Cuisine',
     location: 'Your Location',
     distance: 'Preferred Distance',
     price: 'Price Level',
     experience: 'Dining Experience',
   };
-  const cuisineOptions = ['Italian', 'Mexican'];
-  // This is a filler value. I plan on moving all cuisine stuff to PreferenceForm.js
-  const setCuisine = () => {};
+  const labelsList = {
+    distance: ['1 mile', '5 miles', '10 miles', '25 miles'],
+    price: ['$', '$$', '$$$', '$$$$'],
+    experience: ['Takeout', 'Delivery', 'Dine-In'],
+  };
+  const cuisineOptions = [''];
+
+  const [selectedCuisines, setSelectedCuisines] = useState([]);
+  const [clickedDistanceButtons, setClickedDistanceButtons] = useState([]);
+  const [clickedPriceButtons, setClickedPriceButtons] = useState([]);
+  const [clickedExperienceButtons, setClickedExperienceButtons] = useState([]);
+
+  /**
+   * A callback function that is going to be used to get the updated
+   * map of clicked buttons from the <ButtonGroup> component's on click function.
+   *
+   * The clicked button state will get updated for the corresponding id.
+   */
+  const clickedButtonsCallback = (id, clickedButtonsMap) => {
+    const clickedButtons = [];
+    const labels = labelsList[id];
+    for (const [buttonNumber, isClicked] of Object.entries(clickedButtonsMap)) {
+      if (isClicked) {
+        clickedButtons.push(labels[buttonNumber]);
+      }
+    }
+    if (id === 'distance') {
+      setClickedDistanceButtons(clickedButtons);
+    } else if (id === 'price') {
+      setClickedPriceButtons(clickedButtons);
+    } else {
+      setClickedExperienceButtons(clickedButtons);
+    }
+  };
+
+  /**
+   * Updates the database if the user has changed any of their previous preferences.
+   */
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (!(user && user.isSignedIn())) {
```

Comment:
If there's no user, will user.isSignedIn throw an error? Should we make this check `if (!user || !user.isSignedIn)` ?

Reply:
if `user` is null then this whole boolean statement will short circuit to false so `user.isSignedIn()` won't get called so there won't be an error. if we did the or statement you're suggesting I think there's a chance that `isSignedIn` could be called on a null object which would produce an error.

Reply:
you are indeed correct... i think my brain short circuited when i wrote the above comment :)

Reply:
I think you're both correct 😛 
!(user && user.isSignedIn) will short circuit if user == null because !(false && <anything>) == !(false) 
!user || !user.isSignedIn will also short circuit if user == null because !user == true ==> true || <anything> == true so the second expression doesn;t need to be evaluated

Reply:
Oh wait yes. Ok now my brain is really tripping out 😂

Thank you Surbhi for synthesizing my initial thought!

Reply:
haha indeed - not(A and B) should be equivalent to (not A or not B) so it would be very sketchy if the short circuiting behaviour varied :D

Reply:
ahh i see i see thanks for explaining!! i forgot about the short-circuiting for `true ||`

Reply:
np! as they say - only 3 things in life are certain: taxes, death, and the equivalency of !(A && B) to !A || !B